src/SML.py

```diff
@@ -1116,6 +1114,7 @@ def __init__(self, type: str, x: np.ndarray = None,
     else:
       if model_db is not None:
         model_data = shelve.open(model_db)
+        self.model_db = model_db # need to assign self.model_db
```

Comment:
If the argument `type != "train"` this will return an error later since `self.model_db` is undefined

Reply:
Yes, I agree with that. Your suggested fix is valid.